static/src/js/vuex/library/actions.js

```diff
@@ -1,20 +1,52 @@
 import api from '../../api';
 import constants from '../../constants';
 import transformTextGroupList from './transforms';
+import utils from './utils';
 
 export default {
-  [constants.LIBRARY_LOAD_TEXT_GROUP_LIST]: ({ commit }) => api.getTextGroupList((data) => {
-    const {
-      textGroups,
-      works,
-      texts,
-      textGroupUrns,
-    } = transformTextGroupList(data);
-
-    commit(constants.SET_TEXT_GROUPS, { textGroups, works, texts });
-    commit(constants.SET_TEXT_GROUP_URNS, { textGroupUrns });
-  }),
-
+  [constants.LIBRARY_LOAD_TEXT_GROUP_LIST]: ({ commit }) => {
+    const currentVersion = JSON.parse(localStorage.getItem('libraryDataVersion'));
+    const libraryData = JSON.parse(localStorage.getItem('libraryData'));
+
+    return api.getLibraryInfo((res) => {
+      if (!utils.isCacheValid(res.api_version, currentVersion) || !libraryData) {
+        return api.getTextGroupList((data) => {
+          // add data to localStorage
+          try {
+            localStorage.setItem('libraryData', JSON.stringify(data));
+            const version = {
+              version: res.api_version,
+              date: new Date(),
+            };
+            localStorage.setItem('libraryDataVersion', JSON.stringify(version));
+          } catch (e) {
+            if (utils.isQuotaExceeded(e)) {
+              console.error('localStorage is full!'); // eslint-disable-line no-console
+            }
+          }
+          const {
+            textGroups,
+            works,
+            texts,
+            textGroupUrns,
+          } = transformTextGroupList(data);
+
+          commit(constants.SET_TEXT_GROUPS, { textGroups, works, texts });
+          commit(constants.SET_TEXT_GROUP_URNS, { textGroupUrns });
+        });
+      }
+      const {
+        textGroups,
+        works,
+        texts,
+        textGroupUrns,
+      } = transformTextGroupList(libraryData);
+
+      commit(constants.SET_TEXT_GROUPS, { textGroups, works, texts });
+      commit(constants.SET_TEXT_GROUP_URNS, { textGroupUrns });
+      return true;
```

Comment:
Regarding [my previous comment](https://github.com/scaife-viewer/scaife-viewer/pull/358#discussion_r278294611) about the `return` statement here; I was curious to know if your reason was functional reason to `return true` vs a stylistic one (as the other [library actions](https://github.com/scaife-viewer/scaife-viewer/blob/628ba79a228020621c7f8eb729a19b3df23599f1/static/src/js/vuex/library/actions.js) omit a return statement).

Either way, I don't have a problem merging with this line, I just wanted to know if this is something we should change in our approach to [dispatching actions](https://vuex.vuejs.org/guide/actions.html).

Reply:
I see what you're asking now. Yes, a return is needed there for the calling context (the dispatch) to know that it's complete.